### ISSUE TITLE
AttributeError for matrix conversion

### ISSUE THREAD

**Issue by svalkiers** [networkx/networkx#5410]
Hi,

When trying to convert a graph into a sparse matrix, I run into an `AttributeError` related to the `scipy` package. Within the new 2.7.1 version of `networkx`, the `to_scipy_sparse_array` function looks for a `scipy.sparse.coo_array`, which is a non-existent object. Instead, this should be `scipy.sparse.coo_matrix`.

Here is a little example of the error:

```
m = nx.to_scipy_sparse_array(G)

File ~/miniconda3/envs/clustcr/lib/python3.8/site-packages/networkx/convert_matrix.py:921 in to_scipy_sparse_array
  A = sp.sparse.coo_array((d, (r, c)), shape=(nlen, nlen), dtype=dtype)

AttributeError: module 'scipy.sparse' has no attribute 'coo_array'
```

Python version: 3.8.12
NetworkX version: 2.7.1

Best,
Sebastiaan

**Comment by rossbar** [comment]
Thanks for reporting - the scipy.sparse array interface is only available in the latest scipy version (1.8) so if you are using an older version of scipy you will need to upgrade (e.g. `pip install --upgrade scipy`).

If for whatever reason upgrading is not possible, you can still use the `to_scipy_sparse_matrix` function, which relies on the older scipy.sparse matrix interface, though this will (should) emit a deprecation warning as NX is moving towards array semantics everywhere.

**Comment by rossbar** [comment]
I'm going to close this as resolved - if the issue persists, feel free to reopen with updated info!
